haddock-api/src/Haddock/Interface/Create.hs

```diff
@@ -318,11 +319,11 @@ parseWarning dflags gre w = case w of
 -------------------------------------------------------------------------------
 
 
-mkDocOpts :: Maybe String -> [Flag] -> Module -> ErrMsgM [DocOption]
-mkDocOpts mbOpts flags mdl = do
+mkDocOpts :: SrcSpan -> Maybe String -> [Flag] -> Module -> ErrMsgM [DocOption]
+mkDocOpts loc mbOpts flags mdl = do
   opts <- case mbOpts of
     Just opts -> case words $ replace ',' ' ' opts of
-      [] -> tell ["No option supplied to DOC_OPTION/doc_option"] >> return []
+      [] -> tell [L loc "No option supplied to DOC_OPTION/doc_option"] >> return []
```

Comment:
I know this isn't your doing, but that error message needs to be revised (how old is it!?). The pragma is called `OPTIONS_HADDOCK` and the GHC option is called `-haddock-opts`.

Reply:
Would you prefer that I make this change in a separate commit so the commits remain atomic / minimal?

Reply:
I intend to squash before merging, so it shouldn't make a difference.